app/features/build_runner/build_runner.rb

```diff
@@ -111,22 +132,9 @@ def complete_run(start_time:, artifact_paths: [])
       save_build_status!
     rescue StandardError => ex
       logger.error(ex)
-      duration = Time.now - start_time
-      current_build.duration = duration
-      current_build.status = :failure # TODO: also handle failure
-      save_build_status!
+      fail_build!(start_time: start_time)
```

Comment:
extracted so we could use it in 2 places

Reply:
Ahhh nice catch!!